### ISSUE TITLE
all: transition to a Go module

### ISSUE THREAD

**Issue by mvdan** [mvdan/sh#266]

**Comment by mvdan** [comment]
We rolled this back because apparently there are still many people on Go 1.10. Reopening, to do the change again soon. Perhaps once 1.12 is out, or even sooner.

**Comment by mvdan** [comment]
1.12 is out, so let's move forward with this. Will be merging `master.v3` into `master` with a merge commit in the next few days.

**Comment by mvdan** [comment]
I've just backported some fixes from v3 to v2, and released v2.6.4. I'll create a `v2` branch, much like we did with the `v1` branch, in case we need to do a v2.6.5. That will also make it simpler for users to track the latest backported commits, even if they haven't been tagged.

I'll join the `master.v3` branch with `master` tomorrow. I've pushed some updated installation instructions to the README in that branch, which only work on Go 1.11 and later. Anyone running Go 1.11 or later will be able to switch to one of the two lines of shell in the README to install the version of `shfmt` they want.

For those still on Go 1.10.x, they should consider upgrading to 1.11 or 1.12, or using the download instructions via the standalone `vgo` tool:

```
go get -u golang.org/x/vgo
cd $(mktemp -d); vgo mod init tmp; vgo get mvdan.cc/sh/cmd/shfmt
```

As a last resort to build from source, it's always possible to clone the repository into `$GOPATH`, checkout a specific tag, and `go install ./cmd/shfmt`. And, if building from source is not necessary, there's always the binaries on GitHub and the Docker images at https://hub.docker.com/r/mvdan/shfmt.
